include/qpid/dispatch/router_core.h

```diff
@@ -322,5 +322,5 @@ void qdr_query_free(qdr_query_t *query);
 
 typedef void (*qdr_manage_response_t) (void *context, const qd_amqp_error_t *status, bool more);
 void qdr_manage_handler(qdr_core_t *core, qdr_manage_response_t response_handler);
-
+bool qdr_is_set_manage_handler(qdr_core_t *core);
```

Comment:
I don't think this is needed.  The "outer" agent should register its callbacks once at initialization time.  It is not safe to register new handlers for each request that comes in and it's not necessary to repeatedly register the same handler.